src/students/students.service.ts

```diff
@@ -26,4 +31,95 @@ export class StudentsService {
+    const { entities: students } = await queryBuilder.getRawAndEntities();
+    const pageMetaDto = new PageMetaDto({ itemCount, pageOptionsDto });
+
+    return new PageDto(students, pageMetaDto);
+  }
+
+  private getAllStudentsQuery(pageOptionsDto: PageOptionsDto): SelectQueryBuilder<Student> {
+    return this.dataSource
+      .createQueryBuilder()
+      .from(Student, 'student')
+      .leftJoinAndSelect('student.grades', 'grades')
+      .leftJoinAndSelect('student.profile', 'profile')
+      .skip(pageOptionsDto.skip)
+      .take(pageOptionsDto.take)
+      .select([
+        'student.id',
+        'student.createdAt',
+        'grades.courseCompletion',
+        'grades.courseEngagement',
+        'grades.projectDegree',
+        'grades.teamProjectDegree',
+        'profile.expectedTypeWork',
+        'profile.targetWorkCity',
+        'profile.expectedContractType',
+        'profile.expectedSalary',
+        'profile.canTakeApprenticeship',
+        'profile.monthsOfCommercialExp',
+      ])
+      .orderBy('student.createdAt', pageOptionsDto.order);
+  }
+
+  // eslint-disable-next-line max-lines-per-function
+  public async getOneStudent(studentId: string): Promise<StudentResponse> {
+    const student = await this.getOneStudentQuery(studentId);
+    const studentResponse: StudentResponse = {
+      studentId: student.id,
+      createdAt: student.createdAt,
+      updatedAt: student.updatedAt,
+      details: {
+        profile: {
+          firstName: student.profile.firstName,
+          lastName: student.profile.lastName,
+          githubUsername: student.profile.githubUsername,
+          tel: student.profile.tel,
+          email: student.user.email,
+          bio: student.profile.bio,
+        },
+        grades: {
+          courseCompletion: student.grades.courseCompletion,
+          courseEngagement: student.grades.courseEngagement,
+          projectDegree: student.grades.projectDegree,
+          teamProjectDegree: student.grades.teamProjectDegree,
+        },
+        portfolio: {
+          portfolioUrls: student.profile.portfolioUrls,
+          projectUrls: student.profile.projectUrls,
+          bonusProjectUrls: student.grades.bonusProjectUrls,
+        },
+        employmentExpectations: {
+          expectedTypeWork: student.profile.expectedTypeWork,
+          targetWorkCity: student.profile.targetWorkCity,
+          expectedContractType: student.profile.expectedContractType,
+          expectedSalary: student.profile.expectedSalary,
+          canTakeApprenticeship: student.profile.canTakeApprenticeship,
+          monthsOfCommercialExp: student.profile.monthsOfCommercialExp,
+        },
+        educationAndExperience: {
+          education: student.profile.education,
+          courses: student.profile.courses,
+          workExperience: student.profile.workExperience,
+        },
+      },
+    };
+    return studentResponse;
+  }
+
+  private getOneStudentQuery(studentId: string): Promise<Student> {
```

Comment:
już zrobiłem pobieranie pojedynczego Studenta 
```js
 getStudentById(id: string): Promise<Student> {
    return Student.findOne({ where: { user: { id } }, relations: { user: true, grades: true } });
  }
  ```
  I działa :)

i pewnie wystarczyło się podpiąć do tego, stworzyć endpoint, swaggera i iść spać :)

  ale fakt, może warto korzystać z `dataSource` - podobno lepsze rozwiazanie przy tak sporych połączeniach.
  W takim wypadku powinieneś też od razu przypiąć tą metodę do mojego kodu, odpowiednio poprawić, refakturyzować i usunąć tamtą metodę. Bo teraz mamy 2 metody robiące to samo

Reply:
i oczywiście wszystko przetestować